### PR TITLE
fix: include witness data in estimation

### PR DESCRIPTION
**Pull request by MujkicA**
While estimating to determine the script gas limit we use placeholders for witness data but this assumes that witness data only contains signatures which may not always be the case.
This PR makes sure the existing witnesses are retained before estimating. 

### Checklist
- [ ] I have linked to any relevant issues.
- [ ] I have updated the documentation.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary labels.
- [ ] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [ ] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

File: packages/fuels-core/src/types/transaction_builders/script_dry_runner.rs [hunk 1]
```diff
@@ -78,9 +84,13 @@ impl<R: DryRunner> ScriptDryRunner<R> {
     // Using a `Signature` ensures that the calculated fee includes the fee generated by the witnesses.
     fn add_fake_witnesses(&mut self, tx: &mut fuel_tx::Script) {
         let witness: Witness = Signature::default().as_ref().into();
-        let dry_run_witnesses: Vec<_> = repeat(witness).take(self.num_total_witnesses).collect();
+        let dry_run_witnesses: Vec<_> = repeat(witness)
+            .take(self.num_unresolved_witnesses)
+            .collect();
 
-        *tx.witnesses_mut() = dry_run_witnesses;
+        let predifined_witnesses = std::mem::take(&mut self.predifined_witnesses);
```
Comment (Voxelot) [inline packages/fuels-core/src/types/transaction_builders/script_dry_runner.rs:91]:
```suggestion
        let predefined_witnesses = std::mem::take(&mut self.predefined_witnesses);
```

File: packages/fuels-core/src/types/transaction_builders/script_dry_runner.rs [hunk 2]
```diff
@@ -16,15 +16,21 @@ use crate::{
...
+    predifined_witnesses: Vec<Witness>,
+    num_unresolved_witnesses: usize,
     last_dry_run: Option<DryRun>,
 }
 
 impl<R> ScriptDryRunner<R> {
-    pub fn new(dry_runner: R, num_total_witnesses: usize) -> Self {
+    pub fn new(
+        dry_runner: R,
+        predifined_witnesses: Vec<Witness>,
```
Comment (Voxelot) [inline packages/fuels-core/src/types/transaction_builders/script_dry_runner.rs:27]:
```suggestion
        predefined_witnesses: Vec<Witness>,
```

File: packages/fuels-core/src/types/transaction_builders/script_dry_runner.rs [hunk 3]
```diff
@@ -16,15 +16,21 @@ use crate::{
 
 pub(crate) struct ScriptDryRunner<R> {
     dry_runner: R,
-    num_total_witnesses: usize,
+    predifined_witnesses: Vec<Witness>,
```
Comment (Voxelot) [inline packages/fuels-core/src/types/transaction_builders/script_dry_runner.rs:19]:
```suggestion
    predefined_witnesses: Vec<Witness>,
```

**Review by Br1ght0ne — APPROVED** [review]
Solid work 🚀

**Review by digorithm — APPROVED** [review]
:shipit:

**Review by Voxelot — APPROVED** [review]

**Pull request merged**